spec/rubocop/cop/style/mutable_constant_spec.rb

```diff
@@ -43,6 +43,45 @@
     end
   end
 
+  shared_examples 'freezing all constants' do |o|
```

Comment:
Using `shared_examples` is the right way to go 👍
OTOH, the label or grouping could be improved. "not freezing all constants" is not clear. Something like "enforces mutable literals are frozen" seems more precise, for example.

Reply:
I was trying to make it a readable sentence with **it_behaves_like**. Please do check if this makes sense
_it_behaves_like 'literals that are frozen'_
Also mentioning literals since i would like to include both mutable and immutable literals.

Reply:
There are three categories of values assigned to constants:
1) immutable literals (e.g. `42`, `:symbol`)
2) mutable builtin literals (e.g. `[1, 2, 3]`)
3) general expressions (e.g. `FOO + BAR`, `Set[1, 2, 3]`)

The cop should raise for:
1) never
2) for `none`
3) for `literal` or `none`

I'd like the cop and the specs to reflect this.

Maybe something like "This cop behaves_like:
1) "it_considers_immutable_literals_frozen"
2) "it_considers_builtin_literals_frozen"
3) "it_considers_expressions_in_constants_frozen"

Reply:
Thanks for the detailed comments. I guessed I rushed through the tests a bit, will take a deeper look into this and come back.

Reply:
```ruby
# When EnforcedStyle: literal
# +-------------------+--------------------------------+----------------+
# |                   | Cop offense raised for shareable_constant_value |
# |   Values          |------+---------+---------------+----------------+
# |                   | none | literal | experimental_ | experimental_  |
# |                   |      |         | everything    | copy           |
# +-------------------+------+---------+---------------+----------------+
# |Immutable literals |false |false    |false          |false           |
# |Examples:          |      |         |               |                |
# |42, :symbol        |      |         |               |                |
# +-------------------+------+---------+---------------+----------------+
# |Mutable literals   |true  |false    |false          |false           |
# |Examples:          |      |         |               |                |
# |[1, 2, 3, 4]       |      |         |               |                |
# +-------------------+------+---------+---------------+----------------+
# |General expressions|true  |true     |false          |false           |
# |Examples:          |      |         |               |                |
# |FOO + BAR          |      |         |               |                |
# +-------------------+------+---------+---------------+----------------+
# |Frozen expressions |      |         |               |                |
# |Examples:          |false | false   | false         | false          |
# |(anything).freeze  |      |         |               |                |
# +-------------------+------+---------+---------------+----------------+

# When EnforcedStyle: Strict, Only the following row changes
# +-------------------+------+---------+---------------+----------------+
# |General Expressions|true  |false    |false          |false           |
# |                   |      |         |               |                |
# |Examples: FOO + BAR|      |         |               |                |
# +-------------------+------+---------+---------------+----------------+
```

I am planning to add this comment - Did i summarise correctly? I will go ahead once you are okay with above @marcandre.

Reply:
My observation was more about naming your shared behaviours. If it's not clear to you, we can go ahead as is and eventually refactor.

Reply:
I tried refactoring by breaking it upto smaller chunks - Dint work out.

For now changed the title a bit, will try to refactor this over time.